tests/data/application.json

```diff
@@ -0,0 +1,16 @@
+[
```

Comment:
What is the point of this file? Is there a test?

Reply:
That response will be needed for any tests that run with user_rate_limiting = True. I haven't added aby tests yet, in part because I think the scope of any testing probably merits some discussion but also I didn't want to invest further time now if this was a feature that you didn't intend to include.